CRM/Core/BAO/ConfigSetting.php

```diff
@@ -321,14 +321,23 @@ public static function enableComponent($componentName) {
     }
 
     // get enabled-components from DB and add to the list
-    $enabledComponents = Civi::settings()->get('enable_components');
     $enabledComponents[] = $componentName;
-
     self::setEnabledComponents($enabledComponents);
 
     return TRUE;
   }
 
+  /**
+   * Ensure all components are enabled
+   * @throws CRM_Core_Exception
+   */
+  public static function enableAllComponents() {
+    $allComponents = array_keys(CRM_Core_Component::getComponents());
+    if (Civi::settings()->get('enable_components') != $allComponents) {
```

Comment:
Can we use !==  here

Reply:
![image](https://user-images.githubusercontent.com/2874912/143947592-f8bc3abc-7a1f-4ce3-a1ac-bd8a18eac424.png)

Here we are checking that both arrays contain the same elements, but they don't have to be in the same order.

Reply:
Hmm, interesting:
```
php -r '$a = array(1,2); $b = array(2,1); var_dump($a == $b); var_dump($a === $b);'
bool(false)
bool(false)
```
(php 7.4 at least)

Reply:
I recall stepping through this in the debugger and confirming that the condition is true the first pass but false subsequent test iterations, but wasn't looking too closely.